patroni/__init__.py

```diff
@@ -164,7 +167,28 @@ def shutdown(self):
 
 
 def patroni_main():
-    patroni = Patroni()
+    config = __import__('patroni.config', fromlist=[""])
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+            '--version', action='version',
+            version='%(prog)s {version}'.format(version=__version__)
+            )
+    parser.add_argument(
+            "configfile",
+            nargs="?",
+            default="",
+            help="Patroni may also read the configuration from the {0} environment variable"
+            .format(config.Config.PATRONI_CONFIG_VARIABLE)
+            )
+    args = parser.parse_args()
+    try:
+        conf = config.Config(args.configfile)
+    except ConfigParseError as e:
+        if e.value:
+            print(e.value)
+        parser.print_help()
+        exit(1)
+    patroni = Patroni(conf)
```

Comment:
Patroni could also raise exceptions related to config, for example when initializing DCS module:
```python
  File "/home/akukushkin/git/patroni/patroni/dcs/__init__.py", line 549, in __init__
    self._base_path = re.sub('/+', '/', '/'.join(['', config.get('namespace', 'service'), config['scope']]))
KeyError: 'scope'
```

Looks like the AbstractDCS should also produce some nice readable exceptions.

Reply:
Postgresql and RestApiServer could also raise some exceptions releated to the config.

Reply:
Some stuff could be validated before creation of the Patroni, but unfortunately the config is not complete yet.
The final initialization of the config is done only in the [load_dynamic_configuration()](https://github.com/zalando/patroni/pull/1291/files#diff-0996184a693191991bd4f56d522c4f32R47) method.